src/wait.ts

```diff
@@ -17,41 +17,33 @@ export async function waitForConfirmation(
   // Wait until the transaction is confirmed or rejected, or until 'waitRounds'
   // number of rounds have passed.
 
-  const status = await client.status().do();
-  if (typeof status === 'undefined') {
-    throw new Error('Unable to get node status');
-  }
-  const startRound = status['last-round'] + 1;
-  let currentRound = startRound;
+  let currentRound = 0;
+  let counter = 0;
 
   /* eslint-disable no-await-in-loop */
-  while (currentRound < startRound + waitRounds) {
-    let poolError = false;
+  while (counter <= waitRounds) {
+    let pendingInfo = {};
     try {
-      const pendingInfo = await client.pendingTransactionInformation(txid).do();
-
-      if (pendingInfo['confirmed-round']) {
-        // Got the completed Transaction
-        return pendingInfo;
-      }
-
-      if (pendingInfo['pool-error']) {
-        // If there was a pool error, then the transaction has been rejected
-        poolError = true;
-        throw new Error(`Transaction Rejected: ${pendingInfo['pool-error']}`);
-      }
+      pendingInfo = await client.pendingTransactionInformation(txid).do();
     } catch (err) {
       // Ignore errors from PendingTransactionInformation, since it may return 404 if the algod
       // instance is behind a load balancer and the request goes to a different algod than the
       // one we submitted the transaction to
-      if (poolError) {
-        // Rethrow error only if it's because the transaction was rejected
-        throw err;
-      }
     }
 
-    await client.statusAfterBlock(currentRound).do();
-    currentRound += 1;
+    if (pendingInfo['confirmed-round']) {
+      // Got the completed Transaction
+      return pendingInfo;
+    }
+
+    if (pendingInfo['pool-error']) {
+      // If there was a pool error, then the transaction has been rejected
+      throw new Error(`Transaction Rejected: ${pendingInfo['pool-error']}`);
+    }
+
+    const status = await client.statusAfterBlock(currentRound).do();
```

Comment:
It feels prudent to throw the old error here, i.e.

```
if (typeof status === 'undefined') {
    throw new Error('Unable to get node status');
}
```

Reply:
So the idea of the PR is precisely not to throw an error if there is a temporary network issue before or during an iteration of the loop. If status is undefined the current round will be incremented by 1 in the below line and the loop will have a chance to keep going, the only downside being that the next iteration will happen sooner that it otherwise would have.

Reply:
I can see why it's valuable to not throw an error if there is a temporary network issue, but wouldn't it then be beneficial to include a timeout before the next while loop iteration? Otherwise, as you noted, you're giving no time for the network issue to resolve itself.

Reply:
@jdtzmn Good idea, I've also added a timeout for the first iteration of the loop since `statusAfterBlock` will return almost instantly when `currentRound` is 0.
Could you set the appropriate label for the ci test to pass pls?